### PR TITLE
[#3295] fix(catalog-hive):The Hive Catalog Bug In Multiple Kerberized HMS

### PR DESCRIPTION
**Pull request by theoryxu**
### What changes were proposed in this pull request?

 remove PRINCIPAL -> METASTORE_KERBEROS_PRINCIPAL in the GRAVITINO_CONFIG_TO_HIVE

### Why are the changes needed?

The hive.metastore.kerberos.principal is not the same as kerberos.principal functionally.

Fix: #3295 

### Does this PR introduce _any_ user-facing change?

yes, add the document

### How was this patch tested?

existing test (TestHiveCatalogOperations)

#### test in inner environment as follow:

step 1
Install gravitino in host1, Install HMS1 in host1, and install HMS2 in host2

step 2
create catalog1 for HMS1:

curl -L -X POST 'http://host1:8090/api/metalakes/mk1/catalogs'
-H 'Content-Type: application/json'
-H 'Accept: application/vnd.gravitino.v1+json'
--data-raw '{
"name": "catalog1",
"type": "relational",
"provider": "hive",
"properties": {
"metastore.uris": "thrift://host1:7004",
"kerberos.principal": "hadoop/[host1@EXAMPLE.COM](mailto:host1@EXAMPLE.COM)",
"kerberos.keytab-uri": "/var/krb5kdc/emr.keytab",
"gravitino.bypass.hadoop.security.authentication": "kerberos",
"gravitino.bypass.hive.metastore.kerberos.principal": "hadoop/[_HOST@EXAMPLE.COM](mailto:_HOST@EXAMPLE.COM)",
"gravitino.bypass.hive.metastore.sasl.enabled": true
}
}'

step 3
create catalog1 for HMS1:

curl -L -X POST 'http://host1:8090/api/metalakes/mk1/catalogs'
-H 'Content-Type: application/json'
-H 'Accept: application/vnd.gravitino.v1+json'
--data-raw '{
"name": "catalog2",
"type": "relational",
"provider": "hive",
"properties": {
"metastore.uris": "thrift://host2:7004",
"kerberos.principal": "hadoop/[host1@EXAMPLE.COM](mailto:host1@EXAMPLE.COM)",
"kerberos.keytab-uri": "/var/krb5kdc/emr.keytab",
"gravitino.bypass.hadoop.security.authentication": "kerberos",
"gravitino.bypass.hive.metastore.kerberos.principal": "hadoop/[_HOST@EXAMPLE.COM](mailto:_HOST@EXAMPLE.COM)",
"gravitino.bypass.hive.metastore.sasl.enabled": true
}
}'

step 4
curl -L -X GET 'http://host1:8090/api/metalakes/mk1/catalogs/catalog1/schemas'
success

curl -L -X GET 'http://host1:8090/api/metalakes/mk1/catalogs/catalog2/schemas'
success

### CODE REVIEW

**Comment by qqqttt123** [comment]
Could you modify the document, too? https://github.com/datastrato/gravitino/blob/main/docs/apache-hive-catalog.md

**Comment by qqqttt123** [comment]
The title is too long. The Github will cut off the title. So you need to modify the title. And you should modify the description of the pull request and format the code.
```
./gradlew spotlessApply
```

File: docs/apache-hive-catalog.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ The Hive catalog supports creating, updating, and deleting databases and tables
...
 | `impersonation-enable`                   | Enable user impersonation for Hive catalog.                                                                                                                                                                       | false         | No                           | 0.4.0         |
-| `kerberos.principal`                     | The Kerberos principal for the catalog. You should configure `gravitino.bypass.hadoop.security.authentication` and `gravitino.bypass.hive.metastore.sasl.enabled`if you want to use Kerberos.                     | (none)        | required if you use kerberos | 0.4.0         |
+| `kerberos.principal`                     | The Kerberos principal for the catalog. You should configure `gravitino.bypass.hadoop.security.authentication`, `gravitino.bypass.hive.metastore.kerberos.principal` and `gravitino.bypass.hive.metastore.sasl.enabled`if you want to use Kerberos.                   | (none)        | required if you use kerberos | 0.4.0         |
```
Comment (jerqi) [inline docs/apache-hive-catalog.md:38]: The markdown should align to other rows.

Comment (qqqttt123) [inline docs/apache-hive-catalog.md:38]: The other lines should align to this line.

**Comment by theoryxu** [comment]
> The title is too long. The Github will cut off the title. So you need to modify the title. And you should modify the description of the pull request and format the code.
> 
> ```
> ./gradlew spotlessApply
> ```

done

**Review by qqqttt123 — APPROVED** [review]
LGTM, could you paste the commands which you used?

**Comment by theoryxu** [comment]
> LGTM, could you paste the commands which you used?

step 1
Install gravitino in host1, Install HMS1 in host1, and install HMS2 in host2

step 2
create catalog1 for HMS1:

curl -L -X POST 'http://host1:8090/api/metalakes/mk1/catalogs'
-H 'Content-Type: application/json'
-H 'Accept: application/vnd.gravitino.v1+json'
--data-raw '{
"name": "catalog1",
"type": "relational",
"provider": "hive",
"properties": {
"metastore.uris": "thrift://host1:7004",
"kerberos.principal": "hadoop/host1@EXAMPLE.COM",
"kerberos.keytab-uri": "/var/krb5kdc/emr.keytab",
"gravitino.bypass.hadoop.security.authentication": "kerberos",
"gravitino.bypass.hive.metastore.kerberos.principal": "hadoop/_HOST@EXAMPLE.COM",
"gravitino.bypass.hive.metastore.sasl.enabled": true
}
}'

step 3
create catalog1 for HMS1:

curl -L -X POST 'http://host1:8090/api/metalakes/mk1/catalogs'
-H 'Content-Type: application/json'
-H 'Accept: application/vnd.gravitino.v1+json'
--data-raw '{
"name": "catalog2",
"type": "relational",
"provider": "hive",
"properties": {
"metastore.uris": "thrift://host2:7004",
"kerberos.principal": "hadoop/host1@EXAMPLE.COM",
"kerberos.keytab-uri": "/var/krb5kdc/emr.keytab",
"gravitino.bypass.hadoop.security.authentication": "kerberos",
"gravitino.bypass.hive.metastore.kerberos.principal": "hadoop/_HOST@EXAMPLE.COM",
"gravitino.bypass.hive.metastore.sasl.enabled": true
}
}'

step 4
curl -L -X GET 'http://host1:8090/api/metalakes/mk1/catalogs/catalog1/schemas'
success

curl -L -X GET 'http://host1:8090/api/metalakes/mk1/catalogs/catalog2/schemas'
success

**Comment by theoryxu** [comment]
> LGTM, could you paste the commands which you used?

It has been added to the PR message.

**Comment by qqqttt123** [comment]
@theoryxu Thanks for your contribution! Welcome to Gravitino community! Enjoy your trip!

**Pull request merged**